### PR TITLE
Fix problem of default values populating teamlist.json and write edge test cases

### LINKED ISSUE #121: Write edge test cases for TeamList (Optional<Mentor> and Empty ParticipantList)
Once the fix for #118 has been completed, write tests to ensure that the handling of the Optional Mentor object by Storage is accurate and according to specification. Also need to write tests to ensure that the handling of the empty ParticipantList in a Team object by Storage is accurate and according to specification.

### LINKED ISSUE #122: Fix bug in Storage Tests that results in inadvertent JSON file creation
Running the Storage tests right now causes JSON files to be created in the /data directory. This is undesirable behaviour and severely affects the functionality of the application. A bug fix is urgent.

### PR DESCRIPTION
**Pull request by brianyenna**
###  Description
Describe the pull request and what it does
1. What is this implementing
Previously, the data/teamlist.json file would mysteriously appear and be populated with values from TypicalTeams. This is a simple one-line fix to close #122. 

This PR also fixes #121. 

2. How is it being implemented
For #122, Problem originated in JsonTeamListStorageTest. When running this test, a teamlist.json file is created in the actual /data directory, instead of being created in a temporary directory reserved for testing. 

For #121, the JsonTeamListStorageTest methods have been updated to include these 2 edge cases. 

3. Why is this being implemented
For #122, the pesky bug causes the reading of the teamlist from storage to fail because the participants in the teamlist do not match the actual participants loaded from storage. 

For #121, it is necessary to handle these edge cases as the TeamLists created do not have Mentors and ParticipantLists tagged to them right from initialisation. 

## Checks
- [X] No errors when running the tests
- [X] Build and checkstyle passes
- [X] Run the app and ran 2 commands without failing
- [X] Written the baseline test cases for the PR

## Changelog
- []
- []
- []

### CODE REVIEW

File: src/test/java/seedu/address/storage/JsonTeamListStorageTest.java [hunk 1]
```diff
@@ -61,25 +61,48 @@ public void readTeamList_invalidAndValidTeamList_throwDataConversionException()
         assertThrows(DataConversionException.class, () -> readTeamList("invalidAndValidTeamList.json"));
     }
 
-    @org.junit.jupiter.api.Test
+    @Test
     void readAndSaveTeamList_allInOrder_success() throws Exception {
+        saveAndCompareTeamList(TypicalTeams.getTypicalTeamList(),
+                               TypicalTeams.getTypicalTeams());
```
Comment (hcwong) [inline src/test/java/seedu/address/storage/JsonTeamListStorageTest.java]: Any reason why you chose to do the assert in the helper function instead of in the test function directly?

Comment (brianyenna) [reply src/test/java/seedu/address/storage/JsonTeamListStorageTest.java]: This is to abstract away the details of the assertions, so that I can simply call a single function from all 3 readAndSaveTeamList tests
